util/metric/metric.go

```diff
@@ -205,15 +232,29 @@ func (c *Counter) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Counter.Count())
 }
 
+// Desc implements prometheus.Metric.
+func (c *Counter) Desc() *prometheus.Desc {
+	return c.desc
+}
+
+// Write implements prometheus.Metric.
+func (c *Counter) Write(met *prometheus_model.Metric) error {
+	met.Gauge = &prometheus_model.Gauge{Value: proto.Float64(float64(c.Count()))}
```

Comment:
> Take into account that the Prometheus histograms are cumulative.

That's a good point. We usually have sliding-window ones (by rotating through a bunch of histograms), so we'll want to be smart and count each histogram exactly once, which *does* at a little bit of complexity (we could just record each window which is rotated out from the active position, but that takes as long as the rotation interval, i.e. data is stale at that point).